File: C#/BA1M.cs
using System;
using System.Collections.Generic;

namespace BA1M
{
    class Program
    {
        //A solution to a Rosalind bioinformatics problem
        //Problem Title: Implement NumberToPattern
        //URL: http://rosalind.info/problems/ba1m/
        static void Main(string[] args)
        {
            string NumberToPattern(int number, int k)
            {
                string pattern = "";
                Dictionary<int,string> D = new Dictionary<int,string>();
                D[0] ="A" ;
                D[1] = "C";
                D[2] = "G";
                D[3] = "T";
                int q = number;
                for (int i = 0; i < k; i++)
                {
                    int r = q % 4;
                    q = (int)(q / 4);
                    pattern=pattern + D[r];
                }
                return reverse(pattern);
            }
            string reverse(string text)
            {
                string rev = "";
                int nt = text.Length;
                for (int i = nt - 1; i > -1; i--)
                {
                    rev = rev + text[i];
                }
                return rev;
            }

            string x = "45\n4";
            string[] inlines = x.Split();
            int number = int.Parse(inlines[0]);
            int k = int.Parse(inlines[1]);
            Console.WriteLine(NumberToPattern(number,k));
        }
    }
}

File: Python1/BA1M.py
#!/usr/bin/env python3

"""
A solution to a Rosalind bioinformatics problem
Problem Title: Implement NumberToPattern
URL: http://rosalind.info/problems/ba1m/
"""


def NumberToPattern(number, k):
    pattern = ""
    D = {0: 'A', 1: 'C', 2: 'G', 3: 'T'}
    q = number
    for i in range(0, k):
        r = q % 4
        q = q // 4
        pattern=pattern+D[r]
    return(pattern[::-1])


if __name__ == '__main__':

    #import sys

    #inlines = [x.strip("\n") for x in sys.stdin.readlines()]
    x="45\n4"
    inlines=x.split()
    number = int(inlines[0])
    k = int(inlines[1])

    res = NumberToPattern(number, k)
    print(str(res))
    #sys.stdout.write(str(res))